### DISCUSSION TITLE
Update Product Userfields

### DISCUSSION THREAD

**Discussion by horphi0815** [SebRut/pygrocy#197] [Support] [answered]
Hi,

i would like to understand, how to add values into the user defined fields during 
 grocy.add_generic(entity_type=EntityType.PRODUCTS
and 
grocy.update_generic(entity_type=EntityType.PRODUCTS
BR
Phil

**Comment by SebRut** [comment, marked as answer]
Hey, these methods use the "generic entity interactions" (https://en.demo.grocy.info/api#/Generic%20entity%20interactions) APIs internally and just pass the data object as json to the API.
So to add/update userfields you would just pass a data object with the userfields as a key-value repo like this:

```python
data = {
	"userfields": {
		"myField1" : "myValue1"
	}
}
grocy.add_generic(entity_type=EntityType.PRODUCTS, data)
```

**Reply by horphi0815** [reply]
Great thank you very much. This was my assumption.
BR

Phil

**Reply by horphi0815** [reply]
"SQLSTATE[HY000]: General error: 1 no such column: userfields"
Did you tried this already?
API does not accept it...?

**Reply by SebRut** [reply]
try initialising your grocy instance with `debug=True` for debug logging, and check if the request looks correct. If thats the case please open an issue at grocy.

**Reply by horphi0815** [reply]
Hi @SebRut  again,
i tried the update, but no error returns and no change has been made.
I the API Test it works
https://en.demo.grocy.info/api#/Generic%20entity%20interactions/put_userfields__entity___objectId_
{ "testfield": "2" }

But in PyGrocy not....
Can you help me please? I'm not so familiar to debug it.
```
grocy = Grocy("https://en.demo.grocy.info", "ZGNMF8fTS18BtifyHHpUBcOfcEWTqGoGOyL6WQRiFeGoX3rDe2", port=443,verify_ssl=True)

    try:
        grocy.update_generic(entity_type=EntityType.PRODUCTS, object_id=2, updated_data=
        {
            "userfields":
             {
                 "testfield": "2"
              }
         }
                             )
    except:
        print('error UPDATE')

```
Best regards,
Phil

**Reply by Galapagon** [reply]
Hi everyone, I'm curious what happened to this? It looks like it was even resolved in https://github.com/SebRut/pygrocy/issues/200, but then was closed stagnant and never implemented. 

I ask because I was trying to use userfields to track my chores by room through HomeAssistant and was unable to pull them. https://github.com/custom-components/grocy/issues/217
